test/modules/TestResultCoordinator/Reports/TestSummary.cs

```diff
@@ -1,24 +1,21 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace TestResultCoordinator.Reports
 {
-    using System;
+    using System.Collections.Generic;
     using System.Linq;
     using Microsoft.Azure.Devices.Edge.Util;
 
     class TestSummary
     {
-        internal TestSummary(string testBuildNumber, string deviceId, ITestResultReport[] testResultReports, string blobContainerUri)
+        internal TestSummary(Dictionary<string, string> testInfo, ITestResultReport[] testResultReports, string blobContainerUri)
         {
-            this.TestBuildNumber = Preconditions.CheckNonWhiteSpace(testBuildNumber, nameof(testBuildNumber));
-            this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.TestInfo = Preconditions.CheckNotNull(testInfo, nameof(testInfo));
             this.TestResultReports = Preconditions.CheckNotNull(testResultReports, nameof(testResultReports));
             this.IsPassed = testResultReports.All(r => r.IsPassed);
             this.BlobContainerUri = blobContainerUri ?? string.Empty;
         }
 
-        public string TestBuildNumber { get; }
-
-        public string DeviceId { get; }
```

Comment:
Since device id is getting from environment value, just keep it.  This will be a bit easy to find it when looking for log in blob storage, which part of the path is the device id.